### DISCUSSION TITLE
System.Index in System.Linq.Expressions

### DISCUSSION THREAD

**Discussion by GeroL** [dotnet/runtime#63446] [Q&A]
Hi,

My question is if a System.Index can somehow be used in System.Linq.Expressions like this

`Expression.ArrayAccess(arrayExpr, Expression.Constant(^1));`

Or is this a Roslyn only feature?

**Comment by svick** [comment]
System.Linq.Expressions does not have any explicit support for System.Index. This means that for methods that actually take `Index`, you can use the syntax you suggested. E.g. the following code works:

```c#
var param = Expression.Parameter(typeof(int[]));
Func<int[], Index, Memory<int>> asMemory = MemoryExtensions.AsMemory<int>;
var expression = Expression.Lambda<Func<int[], Memory<int>>>(
    Expression.Call(asMemory.Method, param, Expression.Constant(^1)), param);
var lastItem = expression.Compile().Invoke(new int[] { 1, 2, 3 });
```

Though most of the common operations that use the index syntax in C# aren't like that, so you can't use `Index` there.
